rir/src/compiler/pir/instruction_list.h

```diff
@@ -62,6 +62,8 @@
     V(Subassign2_1D)                                                           \
     V(ForSeqSize)                                                              \
     V(FrameState)                                                              \
+    V(Checkpoint)                                                              \
+    V(ExpectNot)                                                               \
```

Comment:
For the record. I am not sure assume is the proper name. I think the proper name is a mixture between expect and assume. It is true that since that instruction, the opt can assume the condition to hold. But at the same time the semantics of the operation requires to test the condition. It is like a conditional assume. That is usually a guard. Maybe guard is a name to reconsider. Anyway, I followed your suggestion. Since you are going to work on top, maybe you come up to a better name meanwhile.